akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileAndResourceDirectives.scala

```diff
@@ -222,7 +222,16 @@ object FileAndResourceDirectives extends FileAndResourceDirectives {
             ResourceFile(url, entry.getSize, entry.getTime)
           }
         } finally jar.close()
-      case _ ⇒ None
+      case _ ⇒
+        val conn = url.openConnection()
+        try {
+          conn.setUseCaches(false) // otherwise the JDK will keep the JAR file open when we close!
```

Comment:
so it's for when we're inside a jar, and ResourceFile was applied again... I see